v2/futures/websocket_service.go

```diff
@@ -995,50 +996,105 @@ func WsCompositiveIndexServe(symbol string, handler WsCompositeIndexHandler, err
+	AccountUpdate WsAccountUpdate `json:"a"`
+}
+
+type WsUserDataMarginCall struct {
+	CrossWalletBalance  string       `json:"cw"`
+	MarginCallPositions []WsPosition `json:"p"`
+}
+
+type WsUserDataOrderTradeUpdate struct {
+	OrderTradeUpdate WsOrderTradeUpdate `json:"o"`
+}
+
+type WsUserDataTradeLite struct {
+	Symbol          string   `json:"s"`
+	OriginalQty     string   `json:"q"`
+	OriginalPrice   string   //`json:"p"`
+	IsMaker         bool     `json:"m"`
+	ClientOrderID   string   `json:"c"`
+	Side            SideType `json:"S"`
+	LastFilledPrice string   `json:"L"`
+	LastFilledQty   string   `json:"l"`
+	TradeID         int64    `json:"t"`
+	OrderID         int64    `json:"i"`
+}
+
+func (w *WsUserDataTradeLite) fromSimpleJson(j *simplejson.Json) (err error) {
+	w.Symbol = j.Get("s").MustString()
+	w.OriginalQty = j.Get("q").MustString()
+	w.OriginalPrice = j.Get("p").MustString()
+	w.IsMaker = j.Get("m").MustBool()
+	w.ClientOrderID = j.Get("c").MustString()
+	w.Side = SideType(j.Get("S").MustString())
+	w.LastFilledPrice = j.Get("L").MustString()
+	w.LastFilledQty = j.Get("l").MustString()
+	w.TradeID = j.Get("t").MustInt64()
+	w.OrderID = j.Get("i").MustInt64()
+	return nil
+}
+
 func (e *WsUserDataEvent) UnmarshalJSON(data []byte) error {
-	var tmp struct {
-		Event               UserDataEventType     `json:"e"`
-		Time                interface{}           `json:"E"`
-		CrossWalletBalance  string                `json:"cw"`
-		MarginCallPositions []WsPosition          `json:"p"`
-		TransactionTime     int64                 `json:"T"`
-		AccountUpdate       WsAccountUpdate       `json:"a"`
-		OrderTradeUpdate    WsOrderTradeUpdate    `json:"o"`
-		AccountConfigUpdate WsAccountConfigUpdate `json:"ac"`
-	}
-	if err := json.Unmarshal(data, &tmp); err != nil {
+	j, err := newJSON(data)
+	if err != nil {
 		return err
 	}
+	e.Event = UserDataEventType(j.Get("e").MustString())
+	e.Time = j.Get("E").MustInt64()
+	if v, ok := j.CheckGet("T"); ok {
+		e.TransactionTime = v.MustInt64()
+	}
 
-	e.Event = tmp.Event
-	switch v := tmp.Time.(type) {
-	case float64:
-		e.Time = int64(v)
-	case string:
-		parsedTime, err := strconv.ParseInt(v, 10, 64)
-		if err != nil {
-			return err
-		}
-		e.Time = parsedTime
+	eventMaps := map[UserDataEventType]any{
+		UserDataEventTypeMarginCall:          &e.WsUserDataMarginCall,
+		UserDataEventTypeAccountUpdate:       &e.WsUserDataAccountUpdate,
+		UserDataEventTypeOrderTradeUpdate:    &e.WsUserDataOrderTradeUpdate,
+		UserDataEventTypeAccountConfigUpdate: &e.WsUserDataAccountConfigUpdate,
+	}
+
+	switch e.Event {
+	case UserDataEventTypeTradeLite:
+		return e.WsUserDataTradeLite.fromSimpleJson(j)
+	case UserDataEventTypeListenKeyExpired:
+		// noting
 	default:
-		return fmt.Errorf("unexpected type for E: %T", tmp.Time)
+		if v, ok := eventMaps[e.Event]; ok {
+			if err := json.Unmarshal(data, v); err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("unexpected event type: %v", e.Event)
```

Comment:
I'm not sure if I need to return an error here, but I found a lot of unimplemented event types in this code
The latest one I came across recently is COIN_SWAP_ORDER, which is not yet in the documentation
@xyq-c-cpp